### PR TITLE
Windows support (WIP)

### PR DESCRIPTION
**Pull request by kleisauke**
It's work in progress, because I see some errors (with Python 3.6.2 and Windows 10 x64):
<details>
 <summary>Output</summary>

```bash
...............E..........................................................................................................E................E........EFEE......................................
======================================================================
ERROR: test_find_trim (pyvips.tests.test_arithmetic.TestArithmetic)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_arithmetic.py", line 556, in test_find_trim
    left, top, width, height = a.find_trim()
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\vimage.py", line 294, in call_function
    return Operation.call(name, self, *args, **kwargs)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\voperation.py", line 126, in call
    raise Error('no such operation {0}'.format(operation_name))
pyvips.base.Error: no such operation find_trim
  VipsOperation: class "find_trim" not found


======================================================================
ERROR: test_text (pyvips.tests.test_create.TestCreate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_create.py", line 378, in test_text
    im = pyvips.Image.text("Hello, world!")
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\vimage.py", line 131, in call_function
    return Operation.call(name, *args, **kwargs)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\voperation.py", line 126, in call
    raise Error('no such operation {0}'.format(operation_name))
pyvips.base.Error: no such operation text
  VipsOperation: class "text" not found


======================================================================
ERROR: test_jpeg (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 164, in test_jpeg
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmppfrg_k3x.jpg'

======================================================================
ERROR: test_rad (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 607, in test_rad
    self.rad, max_diff = 0)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 89, in save_buffer_tempfile
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmp9rbz3rcj.hdr'

======================================================================
ERROR: test_tiff (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 216, in test_tiff
    self.save_load_file(".tif", "[squash]", self.onebit, 0)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 63, in save_load_file
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmpqdrknwxt.tif'

======================================================================
ERROR: test_vips (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 92, in test_vips
    self.save_load_file(".v", "", self.colour, 0)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 63, in save_load_file
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmp28nbvb_o.v'

======================================================================
FAIL: test_svgload (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 568, in test_svgload
    self.file_loader("svgload", SVG_FILE, svg_valid)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 25, in file_loader
    validate(self, im)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 563, in svg_valid
    self.assertAlmostEqualObjects(a, [79, 79, 132, 255])
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\helpers.py", line 165, in assertAlmostEqualObjects
    self.assertAlmostEqual(x, y, places = places, msg = msg)
AssertionError: 0.0 != 79 within 4 places : 

----------------------------------------------------------------------
Ran 190 tests in 43.545s

FAILED (errors=6, failures=1)
memory: high-water mark 112.16 MB
```
</details>

Not sure if these errors is due to the `tempfile` module. The first 2 errors could be skipped (should we check for libvips 8.6.0 before executing?).

By the way,
Is `libvips 8.6.0-alpha1` available for Windows x64?

### CODE REVIEW

File: pyvips/gvalue.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def get(self):
             result = []
             for i in range(0, pint[0]):
                 vi = array[i]
-                vips_lib.g_object_ref(vi)
+                gobject_lib.g_object_ref(vi)
```
Comment (kleisauke) [inline pyvips/gvalue.py]: If I use `vips_lib` here I see:
```
AttributeError: function/symbol 'g_object_ref' not found in library 'libvips-42.dll': error 0x7f
```

Perhaps this fix should also be applied to lua-vips, see [here](https://github.com/jcupitt/lua-vips/blob/2a3917a7d489bcbeaa324736e089d704d337ac0b/src/vips/gvalue.lua#L214).

Comment (jcupitt) [inline pyvips/gvalue.py]: Good point! Fixed in lua too.

**Comment by jcupitt** [comment]
We should probably check for `find_trim` with something like:

```python
    def test_find_trim(self):
        if pyvips.type_find("VipsOperation", "find_trim") != 0:
            im = pyvips.Image.black(50, 60) + 100
            test = im.embed(10, 20, 200, 300, extend = "white")
            ....
```

ie. make sure `find_trim` is a class before testing.

**Comment by jcupitt** [comment]
There's a w64 build of 8.6 here:

http://www.rollthepotato.net/~john/vips-dev-w64-all-8.6.0.zip

It should have a working `text` as well, I think.

**Comment by jcupitt** [comment]
I'm not sure how to fix the crazy file delete errors, that would need some thinking. 

Maybe put all the temp files in a subdirectory and remove the whole lot at the end after triggering a GC?

File: pyvips/vimage.py [hunk 2]
```diff
@@ -324,11 +324,11 @@ def __exit__(self, type, value, traceback):
     def __getitem__(self, arg):
         if isinstance(arg, slice):
             i = 0
-            if arg.start != None:
+            if arg.start is not None:
```
Comment (jcupitt) [inline pyvips/vimage.py]: Is this a style change, or did you find a bug? It sounds like a good change anyway.

Comment (kleisauke) [inline pyvips/vimage.py]: It's a style change, I received this from PEP8 (a style guide for Python code): 
`E711 comparison to None should be 'if cond is None:'`

See also: https://stackoverflow.com/a/3257957/1480019

**Comment by kleisauke** [comment]
With libvips 8.6.0 I now see:
<details>
 <summary>Output</summary>

```bash
...........................................................................................................................................E........E.EE......................................
======================================================================
ERROR: test_jpeg (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 164, in test_jpeg
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmpq0jirgbx.jpg'

======================================================================
ERROR: test_rad (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 607, in test_rad
    self.rad, max_diff = 0)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 89, in save_buffer_tempfile
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmpjiq15o9g.hdr'

======================================================================
ERROR: test_tiff (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 216, in test_tiff
    self.save_load_file(".tif", "[squash]", self.onebit, 0)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 63, in save_load_file
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmp_9tz__7j.tif'

======================================================================
ERROR: test_vips (pyvips.tests.test_foreign.TestForeign)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 92, in test_vips
    self.save_load_file(".v", "", self.colour, 0)
  File "S:\Users\KleisAuke\Desktop\pyvips\pyvips\tests\test_foreign.py", line 63, in save_load_file
    os.unlink(filename)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\KLEISA~1\\AppData\\Local\\Temp\\tmpc4x41usq.v'

----------------------------------------------------------------------
Ran 190 tests in 43.314s

FAILED (errors=4)
memory: high-water mark 112.16 MB
```
</details>

A few less file delete errors, and find_trim and text is working. I'll incorporate the class check for find_trim and text now.

File: pyvips/vimage.py (hunk 2 above)
Comment (jcupitt) [inline pyvips/vimage.py]: I didn't know about pep8. I guess we should aim to pass it.

```
$ pep8 | wc
   1081    8844   91117
```

Ouch!

Comment (kleisauke) [inline pyvips/vimage.py]: With commit https://github.com/jcupitt/pyvips/pull/5/commits/a81035fb75250372355dc23c35d81a6f6ebf82b6 the whole package passes `pycodestyle` (formerly called `pep8`) and `PyFlakes` (a code analyzer). Tested with `flake8`, which combines `Pyflakes` with style checks against `PEP 8`.

I also removed the `package_index` from ` __init__.py`. (it was solvable with import and referencing the objects using the module name). Running `nosetests` seems to be a bit faster now.

**Comment by jcupitt** [comment]
This PR is getting too large! Let's merge to master before it splits off any further, and fix the remove file errors in another pass.

**Comment by jcupitt** [comment]
Also, this is fantastic Kleis! Docs are the only big thing left now.

**Pull request merged**